src/equality_metrics.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    number_test(eps = eps(Float64))
+
+Generate a test function for number comparison, with absolute tolorance `eps`.
+
+# Examples
+
+```julia
+# string3.txt is "1337"
+@test_reference "references/string3.txt" 1338 # failed
+# set absolute tolorance 10
+@test_reference "references/string3.txt" 1338 test=number_test(10) # pass
+```
+"""
+function number_test(eps = eps(Float64))
+    return (ref, x) -> abs(ref - x) <= eps
+end
```

Comment:
rather than `eps` say `atol`.
Maybe rewrite to use `isapprox`,
and to thus accept `atol` and `rtol` ?


Then call it `approx_equality` ?